src/hardware/serialport/softmodem.cpp

```diff
@@ -718,9 +811,11 @@ void CSerialModem::Timer2(void) {
 				rqueue->addb(txval);
 				//LOG_MSG("Echo back to queue: %x",txval);
 			}
-			if (txval == 0xa) continue; //Real modem doesn't seem to skip this?
-			else if (txval == 0x8 && (cmdpos > 0)) --cmdpos; // backspace
-			else if (txval == 0xd) DoCommand(); // return
+			if (txval == '\n') continue; // Real modem doesn't seem to skip this?
+			else if (txval == '\b') { // backspace
+				if (cmdpos > 0) cmdpos--;
+			}
+			else if (txval == '\r') DoCommand(); // return
```

Comment:
There's no reason to continue statement (`else`) after `continue` here. Also, since you touched these if statements, please reformat them using kernel / K&R style; also I removed comments as the intention is already encoded in `\b`, `\r`.

```suggestion
			if (txval == '\n')
				continue; // Real modem doesn't seem to skip this

			if (txval == '\b') {
				if (cmdpos > 0)
					cmdpos--;
			} else if (txval == '\r') { 
				DoCommand(); 
			} else …
```

Using K&R style makes the code easier to read for us, but also makes it easier to read reports from various tools (static analysis reports, dynamic analysis, etc).

Reply:
Thanks! :)